partners_api/taxi_engine.hpp

```diff
@@ -0,0 +1,87 @@
+#pragma once
+
+#include "storage/index.hpp"
+
+#include "partners_api/taxi_base.hpp"
+
+#include <memory>
+#include <mutex>
+#include <vector>
+
+namespace taxi
+{
+using SuccessfullCallback =
+    std::function<void(ProvidersContainer const & products, uint64_t const requestId)>;
+
+using ErrorCallback = std::function<void(ErrorsContainer const & errors, uint64_t const requestId)>;
+
+class ResultMaker
```

Comment:
Please, explain in the comment why do we need synchronization on this class at all? Do you plan to use it not from UI thread?